src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.FindSymbols.FindReferences;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SymbolMapping;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.InheritanceMargin
+{
+    internal abstract partial class AbstractInheritanceMarginService : IInheritanceMarginService
+    {
+        /// <summary>
+        /// Given the root and the searching span,
+        /// get all the method, event, property and type declaration syntax nodes.
+        /// </summary>
+        protected abstract ImmutableArray<SyntaxNode> GetMembers(SyntaxNode root, TextSpan spanToSearch);
+
+        /// <summary>
+        /// Get the line number for the identifier of declaration node is declared.
+        /// </summary>
+        protected abstract int GetIdentifierLineNumber(SourceText sourceText, SyntaxNode declarationNode);
```

Comment:
see above.  could jsut return the identifier token, and we handle the rest here.